audobject/core/object.py

```diff
@@ -167,12 +167,14 @@ def id(self) -> str:
     @staticmethod
     def from_dict(
             d: typing.Dict[str, typing.Any],
+            stream: typing.IO = None,
             **kwargs,
     ) -> 'Object':
         r"""Create object from dictionary.
 
         Args:
             d: dictionary with variables
+            stream: set to IO stream when dictionary is read from a file
```

Comment:
This I don't get completely. Has the use to set something when reading a file? How can the user set it to "IO stream"? What is an "IO stream"?

Reply:
When the user stores the dictionary to a file, she has to inform the resolver about the location where the dictionary will be stored (this is why we pass the IO stream here, we could also pass the root directory instead if you think that is easier to understand). If not set, the resolver will not be able to create the relative path and the absolute path will be stored. However, usually the user calls `to_yaml()` or `from_yaml()` and then the `stream` is set automatically.

Reply:
So you mean `stream` is the root folder where the `YAML` file (whioch does not exists as we read from a dict) is supposed to be located?

Reply:
Yes, before I was passing the file stream, from which the root folder can be derived. But I know changed it and the root folder is passed directly.